cmd/buildah/bud.go

```diff
@@ -105,12 +119,38 @@ func budCmd(c *cobra.Command, inputArgs []string, iopts budResults) error {
 		pullPolicy = imagebuildah.PullAlways
 	}
 
-	args := make(map[string]string)
+	args := make(argArray)
+
+	p := platforms.DefaultSpec()
+	if c.Flag("platform").Changed {
+		var err error
+		p, err = platforms.Parse(iopts.Platform)
+		if err != nil {
+			return errors.Wrapf(err, "error parsing --platform argument")
+		}
+		args.setArch(p)
+	} else {
+		args.setArch(platforms.DefaultSpec())
+	}
+
 	if c.Flag("build-arg").Changed {
 		for _, arg := range iopts.BuildArg {
 			av := strings.SplitN(arg, "=", 2)
 			if len(av) > 1 {
 				args[av[0]] = av[1]
+				if av[0] == "TARGETPLATFORM" {
```

Comment:
Is TARGETPLATFORM a constant defined somewhere?

Reply:
@rhatdan I'm thinking this PR is dependent on https://github.com/openshift/imagebuilder/pull/142.  @ grooverdan can you confirm?  We may need to hold this PR until the imagebuilder gets merged/vendored.

Reply:
Defined in https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope.

Quite right, should wait for dependent PR https://github.com/openshift/imagebuilder/pull/142. I even checked it works with the version there rather than the slightly different version in this PR.

I'll rebase/revendor/retest on the merge/amendment of https://github.com/openshift/imagebuilder/pull/142.